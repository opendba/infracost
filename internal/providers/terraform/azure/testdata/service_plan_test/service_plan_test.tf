provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "exampleRG1"
  location = "eastus"
}

resource "azurerm_service_plan" "standard_s1" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Windows"

  sku_name     = "S1"
  worker_count = 1
}

resource "azurerm_service_plan" "standard_s2" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Windows"

  sku_name     = "S1"
  worker_count = 5
}

resource "azurerm_service_plan" "premium_v2" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Linux"

  sku_name     = "P1v2"
  worker_count = 10
}

resource "azurerm_service_plan" "basic" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Linux"

  sku_name     = "B2"
  worker_count = 1
}

resource "azurerm_service_plan" "free" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Linux"

  sku_name     = "F1"
  worker_count = 10
}

resource "azurerm_service_plan" "default_capacity" {
  name                = "api-appserviceplan-pro"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  os_type             = "Linux"

  sku_name = "B2"
}
