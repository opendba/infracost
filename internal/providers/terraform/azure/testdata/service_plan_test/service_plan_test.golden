
 Name                                   Monthly Qty  Unit   Monthly Cost (EUR)

 azurerm_service_plan.basic
 └─ Instance usage (B2)                         730  hours              €24.74

 azurerm_service_plan.default_capacity
 └─ Instance usage (B2)                         730  hours              €24.74

 azurerm_service_plan.premium_v2
 └─ Instance usage (P1 v2)                    7,300  hours             €734.78

 azurerm_service_plan.standard_s1
 └─ Instance usage (S1)                         730  hours              €72.75

 azurerm_service_plan.standard_s2
 └─ Instance usage (S1)                       3,650  hours             €363.75

 OVERALL TOTAL (EUR)                                                 €1,220.75
──────────────────────────────────
7 cloud resources were detected:
∙ 5 were estimated
∙ 2 were free
